src/_posts/platform/app/2000-01-01-crash.md

```diff
@@ -1,69 +1,218 @@
 ---
 title: Application Crash
-modified_at: 2016-10-06 00:00:00
+modified_at: 2023-05-17 00:00:00
 tags: app crash recovery
 index: 12
 ---
 
-For various reasons your application can terminate abruptly. Two cases of crash may happen:
+When the platform starts your container(s) (be it after a successful build,
+after a scale operation or after a restart operation), your application enters
+the *run* phase of its [lifespan]({% post_url platform/internals/2000-01-01-container-management %}).
+Unfortunately, bad things may still happen during this phase, which can lead
+your application to crash.
 
-## Boot errors
+Various reasons can explain why your application's lifespan ends up shortened.
+At Scalingo, we distinguish two main kinds of crash:
+**[Boot Errors](#understanding-boot-errors)** and
+**[Runtime Errors](#understanding-runtime-errors)**.
 
-They are detected directly when you deploy your application, if it crashes
-before binding the allocated network port, your deployment fails.
 
-{% note %}
-  The former version of your application __keeps running__.
-{% endnote %}
+## Understanding Boot Errors
+
+**Boot Errors** can only occur when your container is still in its *starting*
+state, **before** entering its *running* state. At this moment, your build is
+successful, but your deployment is still ongoing and the platform doesn't
+consider your application as *running* yet.
+
+These errors are thrown by the platform when it detects that your application
+doesn't behave as expected.
+
+There are 3 kinds of Boot Errors:
+[Start Errors](#understanding-start-errors),
+[Timeout Errors](#understanding-timeout-errors) and
+[Hook Errors](#understanding-hook-errors).
+
+### Understanding Start Errors
 
-## Runtime crashes
+The **Start Error** is the default kind of Boot Error. It is thrown as soon as
+an unmanaged error is detected and caught by the platform. It can occur at any
+moment, as long as your deployment isn't *running* yet.
```

Comment:
```suggestion
The **Start Error** is the default kind of Boot Error. It is thrown as soon as
an unmanaged error is detected and caught by the platform. It can occur at any
moment, as long as your app isn't *running* yet.
```